app/attachments.js

```diff
@@ -167,3 +169,63 @@ exports.createAbsolutePathGetter = rootPath => relativePath => {
   }
   return normalized;
 };
+
+//      getTempPath :: AbsolutePath -> AbsolutePath
+exports.getTempPath = userDataPath => {
+  if (!isString(userDataPath)) {
+    throw new TypeError("'userDataPath' must be a string");
+  }
+  return path.join(userDataPath, TEMPPATH);
+};
+
+//      createWriterForTempAttachment :: AttachmentsPath -> ArrayBuffer -> Ext ->
+//                            { Filename , ArrayBuffer }
+exports.createWriterForTempAttachment = root => {
+  if (!isString(root)) {
+    throw new TypeError("'root' must be a path");
+  }
+
+  return async (arrayBuffer) => {
+    fse.ensureDirSync(root);
+
+    const buffer = Buffer.from( new Uint8Array(arrayBuffer) );
+    const filename = `${uuidv4()}`;
+    const tempFilePath = path.join(root, filename);
+
+    fse.writeFileSync(tempFilePath, buffer);
+
+    const newBuffer = fse.readFileSync(tempFilePath);
+    const returnArrayBuffer = new ArrayBuffer(newBuffer.length);
+    const view = new Uint8Array(returnArrayBuffer);
+
+    newBuffer.forEach((value, index) => {
+      view[index] = value;
+    });
+
+    return { filename, arrayBuffer: returnArrayBuffer };
+  };
+};
+
+//      createDeleterForTempAttachment :: AttachmentsPath -> null
+exports.createDeleterForTempAttachment = root => {
+  if (!isString(root)) {
+    throw new TypeError("'root' must be a path");
+  }
+
+  return async filename => {
+    fse.ensureDirSync(root);
+
+    if(filename) {
```

Comment:
Kinda feel like this should be two different calls to prevent full-directory deletion if the passed-in parameter is falsey. Like an empty string.